libraries/joomla/table/table.php

```diff
@@ -1520,6 +1674,15 @@ public function move($delta, $where = '')
 	 */
 	public function publish($pks = null, $state = 1, $userId = 0)
 	{
+		// Pre-processing by observers
+		$event = AbstractEvent::create('onTableBeforePublish', [
+			'subject'	=> &$this,
+			'pks'		=> $pks,
+			'state'		=> $state,
+			'userId'	=> $userId,
+		]);
+		$this->getDispatcher()->dispatch('onTableBeforePublish', $event);
+
 		// Sanitize input
 		$userId = (int) $userId;
```

Comment:
I think this needs to move up - if the user id isn't an `int` in the event then we're throwing exceptions